apis/python/src/tiledbsoma/_collection.py

```diff
@@ -436,7 +436,12 @@ def __getitem__(self, key: str) -> CollectionElementType:
             timestamp = self.tiledb_timestamp_ms
 
             try:
-                wrapper = _tdb_handles.open(uri, mode, context, timestamp)
+                soma_type = None
+                if entry.entry.wrapper_type == _tdb_handles.ArrayWrapper:
+                    soma_type = "SOMAArray"
+                elif entry.entry.wrapper_type == _tdb_handles.GroupWrapper:
+                    soma_type = "SOMAGroup"
```

Comment:
These seem like they could be members of the `_tdb_handles.[whatever]Wrapper` classes, eliminating the need for this test.

Would `storage_type` or `tiledb_type` be a better name for this? I would expect `soma_type` to refer to the name of the type of SOMA object that is being opened (like `SOMADataFrame`).